adapters/rhythmone/rhythmone.go

```diff
@@ -0,0 +1,218 @@
+	if len(request.Imp) == 0 {
+		err := &errortypes.BadInput{
+			Message: "No imp data provided in the bid request.",
+		}
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	validImpDataExists := false
+
+	for _, imp := range request.Imp {
+		if imp.Banner != nil {
+			validImpDataExists = true
+		} else if imp.Video != nil {
+			validImpDataExists = true
+		} else {
+			err := &errortypes.BadInput{
+				Message: fmt.Sprintf("RhythmOne only supports banner and video imps. Ignoring imp id=%s", imp.ID),
+			}
+			errs = append(errs, err)
+		}
+	}
+
+	if !validImpDataExists {
+		err := &errortypes.BadInput{
+			Message: fmt.Sprintf("No valid imp data in the bid request"),
+		}
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	var uri string
+	request, uri, errs = a.preProcess(request, errs)
+	if request != nil {
+		reqJSON, err := json.Marshal(request)
+		if err != nil {
+			errs = append(errs, err)
+			return nil, errs
+		}
+		if uri != "" {
+			headers := http.Header{}
+			headers.Add("Content-Type", "application/json;charset=utf-8")
+			headers.Add("Accept", "application/json")
+			return []*adapters.RequestData{{
+				Method:  "POST",
+				Uri:     uri,
+				Body:    reqJSON,
+				Headers: headers,
+			}}, errs
+		}
+	}
+	return nil, errs
+}
+
+func (a *RhythmoneAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
+	}
+	var bidResp openrtb.BidResponse
+	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("bad server response: %d. ", err),
+		}}
+	}
+
+	var errs []error
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(5)
+
+	for _, sb := range bidResp.SeatBid {
+		for i := range sb.Bid {
+			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+				Bid:     &sb.Bid[i],
+				BidType: getMediaTypeForImp(sb.Bid[i].ImpID, internalRequest.Imp),
+			})
+		}
+	}
+	return bidResponse, errs
+}
+
+func getMediaTypeForImp(impId string, imps []openrtb.Imp) openrtb_ext.BidType {
+	mediaType := openrtb_ext.BidTypeBanner
+	for _, imp := range imps {
+		if imp.ID == impId {
+			if imp.Banner != nil {
+				mediaType = openrtb_ext.BidTypeBanner
+			} else if imp.Video != nil {
+				mediaType = openrtb_ext.BidTypeVideo
+			}
+			if imp.Banner != nil && imp.Video != nil {
+				mediaType = openrtb_ext.BidTypeBanner
```

Comment:
This `if imp.Banner != nil && imp.Video != nil {` block is a no-op, and can be deleted.

If both are non-nil, the `if imp.Banner != nil {` block will already have set this to BidTypeBanner.